### DISCUSSION TITLE
arduino nano rp2040 scoppy firmware

### DISCUSSION THREAD

**Discussion by AlexanderCaceres** [fhdm-dev/scoppy#112] [Q&A] [answered]
Hi

I got a arduino nano rp2040 instead of a pi pico, I'm interested on using the scoppy firmware I notice the GPIO26 and GPIO27 is available, as shown on this [image] (https://docs.arduino.cc/static/a4d65079ebe3058e37153eb0f93890a8/ABX00053-pinout.png)

have any of you tried to use the firmware on the "arduino nano rp2040"  ? if yes any adjustment we need to do?

**Comment by fhdm-dev** [comment, marked as answer]
I haven't tried using Scoppy with an Arduino Nano RP2040 Connect but the non-wifi version of the firmware might work. I suspect you'll need to reassign some of the GPIOs. You do this in the Scoppy app by tapping the connection badge at the lower left of the screen and then tap 'Connected Device' and then 'Firmware Settings' and then 'GPIOs'.

The list of GPIOs used by the Scoppy firmware (by default) can be found here: https://oscilloscope.fhdm.xyz/wiki/GPIOs
From the pinout diagram of the Arduino it looks like GPIOs 2,3,5,14 and 22 are not available so I assume they are used internally by the Arduino. You might want to disable these (by setting them to NONE in the Scoppy firmware settings). 

Before installing the Scoppy firmware it might be worth checking what is connected to GPIO14 and GPIO22 because the Scoppy firmware will initialise these as outputs (until you get a chance to set them to NONE). I guess there is the potential for this to cause problems if other hardware on the board expects them to be inputs.

**Reply by AlexanderCaceres** [reply]
Thanks, I will try it
